### PR TITLE
Fix: Added Auto release actions using release-please-actions

### PR DESCRIPTION
**Pull request by ssuish**
# Description

This action will automate releases in release branches that has this pattern: `[0-9][0-9][0-9][0-9]-*` i.e. `release/2023-01` or `release/2023-1`, and automatically deploy it on the server as prod. Upon implementing this feature, **ALL COMMITS** should follow 
[conventional-commits](https://www.conventionalcommits.org/en/v1.0.0/) commits convention in order to be tracked in `CHANGELOG.md`. By default, the versioning strategy of conventional-commits follows [semver](https://semver.org), I think that's the standard. This is related to prod, please review with caution. Thank you! :wave: 

Fixes # (issue)

Initial Infrastructure

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

**Test Configuration**:
* Firmware version:
* Hardware:
* Toolchain:
* SDK:

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by Jedddy — DISMISSED** [review]
lgtm!!

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+        uses: garygrossgarten/github-action-ssh@release
+        with:
+          host: ${{ secrets.SERVER_HOST }}
+          username: ${{ secrets.SERVER_USERNAME }}
+          privateKey: ${{ env.SERVER_SSH_KEY }}
+          command: |
+            cd pph
+            git pull
+            docker stop progphil-bot-prod || true && docker rm progphil-bot-prod || true
+            docker rm progphil-bot-prod
```
Comment (CunningBard) [inline .github/workflows/release.yml:59]: Running docker rm on `progrphil-bot-prod` again?

Comment (CunningBard) [reply .github/workflows/release.yml:59]: I don't know why you're doing it again, so I'm just asking

Comment (ssuish) [reply .github/workflows/release.yml:59]: mb mb, line 58 alone could check and remove existing container. The redundant docker rm is just there to test if it's really removed.

**Review by Jedddy — APPROVED** [review]
lgtm!

**Review by vouchard — APPROVED** [review]
Lgtm

**Review by CunningBard — APPROVED** [review]
LGTM, Thanks For answering my question!

**Pull request merged**